### DISCUSSION TITLE
Improve documentation for running docker image with sqlite backend

### DISCUSSION THREAD

**Discussion by dazzag24** [requarks/wiki#3514] [Error / Bug Report]
Improve documentation for running docker image with sqlite backend.  The docs had no indication of how to create a volume to host the sqlite file.

```
mkdir $HOME/wiki.js
docker run -d -p 8080:3000 --name wiki --restart unless-stopped -v $HOME/wiki.js:/root -e DB_TYPE=sqlite -e DB_FILEPATH=/root/wiki.sqlite requarks/wiki:2
```

**Comment by nlfiedler** [comment]
Running into this myself today, no idea how to get the container to run as the user that has permissions to access the volume. I do this with other containers, but it fails miserably with the requarks/wiki:2 image. Would be nice to see the Dockerfile definition for this image so that I can build my own.

**Comment by dazzag24** [comment]
Did the example I gave not work?
It works for me.

**Comment by nlfiedler** [comment]
Sorry, no, it didn't. I tried a lot of things, and gave up on a writable volume and instead mapped a ro volume with SSH keys to replicate to GitHub. Tried local storage and SQLite, nothing involving a writable volume would work because of file permissions. Seems like the user in the wiki container is `node` but could not get any sort of setup to work. Either it couldn't write to the host FS or it failed to access its own files (e.g. `package.json`) within the container.

**Comment by danielehrhardt** [comment]
Same here
```
2020-12-16T16:57:17.294Z [MASTER] [32minfo[39m: Connecting to database...
2020-12-16T16:57:17.295Z [MASTER] [31merror[39m: Database Connection Error: SQLITE_CANTOPEN undefined:undefined
```

**Comment by Ertuit** [comment]
I figured out, after maybe 2-3h of trying.
You need to put the` -u="root"` . 
The complete command look like that :
`docker run -d -p 8080:3000 -u="root" --name wiki --restart unless-stopped -v $HOME/wiki.js:/root -e DB_TYPE=sqlite -e DB_FILEPATH=/root/wiki.sqlite requarks/wiki:2`
